File: src/Cayita.Javascript.Demo/modules/DemoPanels.js
(function() {
	////////////////////////////////////////////////////////////////////////////////
	// Modals.DemoPanels
	var $DemoPanels = function() {
	};
	$DemoPanels.execute = function(parent) {
		$(parent).append(Cayita.UI.StringExtensions.header('Panels', 2));
		(new Cayita.UI.Div.$ctor2(null, function(div) {
			div.className = 'bs-docs-example';
			div.style.paddingLeft = '40px';
			$(div).append(Cayita.UI.Div.createContainer$1(div, function(cf) {
				Cayita.UI.Div.createRow$1(cf, function(rf) {
					new Cayita.UI.Div.$ctor2(rf, function(ld) {
						ld.className = 'span5';
						var apps = (new Cayita.UI.Panel()).caption('Apps').closable(false).resizable(false);
						new Cayita.UI.Div(function(id) {
							id.className = 'c-icons';
							$(id).append('<style>img {height: 40px;}  .c-icon {height: 95px;}</style>');
							var $t1 = $Modals_App.getApps();
							for (var $t2 = 0; $t2 < $t1.length; $t2++) {
								var app = { $: $t1[$t2] };
								new Cayita.UI.Anchor.$ctor3(id, ss.mkdel({ app: app }, function(a) {
									a.className = 'c-icon';
									new Cayita.UI.Image.$ctor2(a, ss.mkdel({ app: this.app }, function(img) {
										img.src = this.app.$.get_icon();
										img.className = 'img-rounded';
									}));
									$(a).on('click', ss.mkdel({ app: this.app }, function(ev) {
										ev.preventDefault();
										Alertify.log.info(this.app.$.get_tittle(), 5000);
									}));
									new Cayita.UI.Span.$ctor2(a, ss.mkdel({ app: this.app }, function(sp) {
										sp.className = 'c-icon-label';
										sp.innerHTML = this.app.$.get_tittle();
									}));
								}));
							}
							apps.append$4(id);
						});
						apps.render(ld);
						var pn = (new Cayita.UI.Panel()).caption('Demo Panel').closable(false).resizable(false).draggable(false);
						new Cayita.UI.Button(function(bt) {
							$(bt).text('Grey background');
							$(bt).on('click', function(evt) {
								pn.body().style.backgroundColor = 'grey';
							});
							pn.append$4(bt);
						});
						new Cayita.UI.Button(function(bt1) {
							$(bt1).text('White background');
							$(bt1).on('click', function(evt1) {
								pn.body().style.backgroundColor = 'white';
							});
							pn.append$4(bt1);
						});
						new Cayita.UI.Button(function(bt2) {
							$(bt2).text('Collapse');
							$(bt2).on('click', function(evt2) {
								pn.collapse();
							});
							pn.append$4(bt2);
						});
						pn.render(ld);
					});
					new Cayita.UI.Div.$ctor2(rf, function(ld1) {
						ld1.className = 'span5';
						var coyote = (new Cayita.UI.Panel()).caption('El Coyote').closable(false).render(ld1).draggable(false).resizable(false);
						new Cayita.UI.Div(function(d) {
							d.className = 'span2';
							new Cayita.UI.Image.$ctor2(d, function(i) {
								i.src = 'img/coyote.jpg';
								i.style.height = '20%';
							});
							coyote.append$4(d);
						});
						new Cayita.UI.Div(function(d1) {
							d1.className = 'span10';
							$(d1).append('<i><b>El <a href=\'https://es.wikipedia.org/wiki/Coyote\' title=\'Coyote\' target=\'_blank\'>Coyote</a> y el <a href=\'https://es.wikipedia.org/wiki/Geococcyx_californianus\' title=\'Geococcyx californianus\' target=\'_blank\'>Correcaminos</a></b></i> (<i><b>Wile E. Coyote</b> and the <b>Road Runner</b></i>) son los personajes de una serie <a href=\'https://es.wikipedia.org/wiki/Estados_Unidos\' title=\'Estados Unidos\' target=\'_blank\'>estadounidense</a> de <a href=\'https://es.wikipedia.org/wiki/Dibujos_animados\' title=\'Dibujos animados\' target=\'_blank\'>dibujos animados</a> creada en el año de <a href=\'https://es.wikipedia.org/wiki/1949\' title=\'1949\' target=\'_blank\'>1949</a> por el animador <a href=\'https://es.wikipedia.org/wiki/Chuck_Jones\' title=\'Chuck Jones\' target=\'_blank\'>Chuck Jones</a> para <a href=\'https://es.wikipedia.org/wiki/Warner_Brothers\' title=\'Warner Brothers\' target=\'_blank\'>Warner Brothers</a>. Chuck Jones se inspiró para crear a estos personajes en un libro de <a href=\'https://es.wikipedia.org/wiki/Mark_Twain\' title=\'Mark Twain\' target=\'_blank\'>Mark Twain</a>, titulado <i>Roughin It</i>, en el que Twain denotaba que los coyotes hambrientos podrían cazar un correcaminos.<br/><a href=\'https://es.wikipedia.org/wiki/El_Coyote_y_el_Correcaminos\' title=\'Coyote\' target=\'_blank\'>El Coyote (wikipedia)</a> ');
							coyote.append$4(d1);
						});
						var tbp = (new Cayita.UI.Panel()).caption('Table').closable(false).draggable(false).resizable(false);
						new Cayita.UI.HtmlTable.$ctor1(function(t) {
							t.className = 'table table-striped table-hover table-condensed';
							new Cayita.UI.TableHeader.$ctor1(t, function(h) {
								new Cayita.UI.TableRow.$ctor1(h, function(tr) {
									new Cayita.UI.TableCell.$ctor3(tr, function(td) {
										$(td).text('Title');
									});
									new Cayita.UI.TableCell.$ctor3(tr, function(td1) {
										$(td1).text('URL');
									});
								});
							});
							var $t3 = $Modals_App.getApps();
							for (var $t4 = 0; $t4 < $t3.length; $t4++) {
								var app1 = { $: $t3[$t4] };
								new Cayita.UI.TableRow.$ctor1(t, ss.mkdel({ app1: app1 }, function(tr1) {
									new Cayita.UI.TableCell.$ctor3(tr1, ss.mkdel({ app1: this.app1 }, function(td2) {
										$(td2).text(this.app1.$.get_tittle());
									}));
									new Cayita.UI.TableCell.$ctor3(tr1, ss.mkdel({ app1: this.app1 }, function(td3) {
										$(td3).text(this.app1.$.get_icon());
									}));
								}));
							}
							tbp.append$4(t);
						});
						tbp.render(ld1);
					});
				});
			}).element());
			$(div).append('');
		})).appendTo$1(parent);
		(new Cayita.UI.Div.$ctor2(null, function(div1) {
			div1.className = 'bs-docs-example';
			var i1 = 1;
			new Cayita.UI.Button.$ctor2(div1, function(bt3) {
				$(bt3).text('Window I');
				$(bt3).on('click', function(evt3) {
					(new Cayita.UI.Panel()).caption('Window ' + (i1++).toString()).left((i1 * 5).toString() + 'px').top((i1 * 15).toString() + 'px').width('300px').height('100px').overlay(true).render(null);
				});
			});
			$(div1).append('');
		})).appendTo$1(parent);
		(new Cayita.UI.Div.$ctor2(null, function(div2) {
			div2.className = 'bs-docs-example';
			new Cayita.UI.Button.$ctor2(div2, function(bt4) {
				$(bt4).text('Window II');
				$(bt4).on('click', function(evt4) {
					(new Cayita.UI.Panel()).caption('Custom Close Icon and Handler').overlay(true).left('20px').top('200px').width('auto').closeIconClass('icon-th-large').closable(true).closeIconHandler(function(p) {
						p.caption('Icon Close Changed !!! ');
						p.closeIconClass('icon-remove-circle');
						p.closeIconHandler(function(pn1) {
							pn1.close();
						});
						p.height('400px');
					}).onCloseHandler(function(p1) {
						Alertify.log.info(Cayita.UI.StringExtensions.header('panel closed ', 3).outerHTML, 5000);
					}).append$3(new Cayita.UI.Button(function(b) {
						$(b).text('Click me');
						b.style.width = '100%';
						b.style.height = '100%';
						$(b).on('click', function(ev1) {
							Alertify.log.info('button clicked!!!', 5000);
						});
					})).render(null);
				});
			});
			$(div2).append('');
		})).appendTo$1(parent);
		(new Cayita.UI.Div.$ctor2(null, function(div3) {
			div3.className = 'bs-docs-example';
			new Cayita.UI.Button.$ctor2(div3, function(bt5) {
				$(bt5).text('Window III');
				$(bt5).on('click', function(evt5) {
					var error = new Cayita.UI.Paragraph.$ctor1(function(p2) {
						p2.style.color = 'red';
						$(p2).append((new Cayita.UI.Icon.$ctor1(function(i2) {
							i2.className = 'icon-minus-sign';
							i2.style.marginTop = '8px';
							i2.style.marginRight = '8px';
						})).element());
						$(p2).append(' panel was closed ');
					});
					var pn2 = (new Cayita.UI.Panel()).caption('No Closable No Collapsible').overlay(true).left('30px').top('400px').width('auto').closable(false).collapsible(false).onCloseHandler(function(p3) {
						error.logInfo(5000);
					});
					pn2.append$3(new Cayita.UI.Button(function(b1) {
						$(b1).text('Click me');
						b1.style.width = '100%';
						b1.style.height = '100%';
						$(b1).on('click', function(be) {
							pn2.closable(true);
							pn2.collapsible(true);
							pn2.caption('Now Closable and Collapsible');
							b1.disabled = true;
						});
					}));
					pn2.render(null);
				});
			});
			$(div3).append('');
		})).appendTo$1(parent);
		;
	};
	////////////////////////////////////////////////////////////////////////////////
	// Modals.App
	var $Modals_App = function() {
		this.$1$TittleField = null;
		this.$1$IconField = null;
	};
	$Modals_App.prototype = {
		get_tittle: function() {
			return this.$1$TittleField;
		},
		set_tittle: function(value) {
			this.$1$TittleField = value;
		},
		get_icon: function() {
			return this.$1$IconField;
		},
		set_icon: function(value) {
			this.$1$IconField = value;
		}
	};
	$Modals_App.getApps = function() {
		var $t1 = [];
		var $t2 = new $Modals_App();
		$t2.set_tittle('Calculator');
		$t2.set_icon('img/calculator.png');
		ss.add($t1, $t2);
		var $t3 = new $Modals_App();
		$t3.set_tittle('Control Panel');
		$t3.set_icon('img/control.png');
		ss.add($t1, $t3);
		var $t4 = new $Modals_App();
		$t4.set_tittle('Firewall Settings');
		$t4.set_icon('img/firewall.png');
		ss.add($t1, $t4);
		var $t5 = new $Modals_App();
		$t5.set_tittle('Spreadsheet');
		$t5.set_icon('img/calc.png');
		ss.add($t1, $t5);
		var $t6 = new $Modals_App();
		$t6.set_tittle('Mail');
		$t6.set_icon('img/mail.png');
		ss.add($t1, $t6);
		var $t7 = new $Modals_App();
		$t7.set_tittle('Jack Sparrow Navigator');
		$t7.set_icon('img/web.png');
		ss.add($t1, $t7);
		var $t8 = new $Modals_App();
		$t8.set_tittle('MonoDevelop');
		$t8.set_icon('img/monodevelop.png');
		ss.add($t1, $t8);
		var $t9 = new $Modals_App();
		$t9.set_tittle('Tomboy');
		$t9.set_icon('img/tomboy.png');
		ss.add($t1, $t9);
		var $t10 = new $Modals_App();
		$t10.set_tittle('Skype');
		$t10.set_icon('img/skype.png');
		ss.add($t1, $t10);
		var a = $t1;
		return a;
	};
	ss.registerClass(global, 'DemoPanels', $DemoPanels);
	ss.registerClass(global, 'Modals.App', $Modals_App);
})();

File: src/Cayita.Demo.Panels/DemoPanels.cs
using System;
using System.Runtime.CompilerServices;
using jQueryApi;
using System.Collections.Generic;

namespace Cayita.Demo
{
	[IgnoreNamespace]
	public class DemoPanels
	{
		public DemoPanels ()
		{
		}

		public static void Execute(Atom parent)
		{
			var code = new Div ();

			new Div (d=>{
				d.ClassName="bs-docs-example";

				UI.CreateContainer( d, ct=>{

					UI.CreateRowFluid(ct, r=>{

						new Div(r, sp=>{
							sp.ClassName="span5";

							var p1 =CreatePanel("Apps");

							new Div(pi=>{
								pi.ClassName="c-icons";
								pi.Append("<style>img {height: 40px;}  .c-icon {height: 95px;}</style>");

								foreach( var app in GetApps()){
									new ImgPicture(pi, img=>{
										img.Img.Src=app.Icon;
										img.Text=app.Title;
										img.Clicked+= (e) =>{e.PreventDefault(); app.Title.LogInfo();};
									});
								}
								p1.Add(pi);
							});

							var p2 =CreatePanel("Demo");

							var color = new TextField();
							var bb = new Button("Change background");
							bb.Clicked+= (e) => p2.Body.Style.BackgroundColor= color.Value;
							var bc = new Button("collapse");
							bc.Clicked+= (e) =>  p2.Collapse();

							p2.Add(color);
							p2.Add(bb);
							p2.Add(bc);

							sp.JQuery.Append(p1).Append(p2);


						});

						new Div(r, sp=>{
							sp.ClassName="span5";

							var p1 =CreatePanel("El Coyote");

							new Div( cy=>{  
								cy.ClassName="span2";  
								new Image(cy,i=>{  
									i.Src="img/coyote.jpg";  
									i.Style.Height="100px";  
								});  
								p1.Add(cy);
							});  

							new Div( cy=>{  
								cy.ClassName="span10";  
								cy.Append(@"<i><b>El <a href='https://es.wikipedia.org/wiki/Coyote' title='Coyote' target='_blank'>Coyote</a> y el <a href='https://es.wikipedia.org/wiki/Geococcyx_californianus' title='Geococcyx californianus' target='_blank'>Correcaminos</a></b></i> (<i><b>Wile E. Coyote</b> and the <b>Road Runner</b></i>) son los personajes de una serie <a href='https://es.wikipedia.org/wiki/Estados_Unidos' title='Estados Unidos' target='_blank'>estadounidense</a> de <a href='https://es.wikipedia.org/wiki/Dibujos_animados' title='Dibujos animados' target='_blank'>dibujos animados</a> creada en el año de <a href='https://es.wikipedia.org/wiki/1949' title='1949' target='_blank'>1949</a> por el animador <a href='https://es.wikipedia.org/wiki/Chuck_Jones' title='Chuck Jones' target='_blank'>Chuck Jones</a> para <a href='https://es.wikipedia.org/wiki/Warner_Brothers' title='Warner Brothers' target='_blank'>Warner Brothers</a>. Chuck Jones se inspiró para crear a estos personajes en un libro de <a href='https://es.wikipedia.org/wiki/Mark_Twain' title='Mark Twain' target='_blank'>Mark Twain</a>, titulado <i>Roughin It</i>, en el que Twain denotaba que los coyotes hambrientos podrían cazar un correcaminos.  
<a href='https://es.wikipedia.org/wiki/El_Coyote_y_el_Correcaminos' title='Coyote' target='_blank'>El Coyote (wikipedia)</a> ");  
								p1.Add(cy);
							});  


							var p2 =CreatePanel("Table");
							var tb = new Table<App>("title");
							tb.Load(GetApps());
							p2.Add(tb);
							sp.JQuery.Append(p1).Append(p2);


						});

					});

				});


				var wn=1;  

				new Button(d, bt=>{
					bt.Text="Window I";  
					bt.Clicked+=evt=>{  
						new Panel( new PanelOptions{Overlay=true}){
							Caption="Window " +(wn++).ToString(),
							Left=(wn*5).ToString()+"px",  
							Top=(wn*15).ToString()+"px",  
							Width="300px",  
							Height="100px",

						}.Show();  
					};
				});


				new Button(d, bt=>{
					bt.Text="Window II";  
					bt.Clicked+=evt=>{  
						var panel= new Panel( new PanelOptions{
							Overlay=true,
							Caption="Custom Close Icon and Handler",
							Left="200px",  
							Top="300px",  
							Width="auto",  
							CloseIconClass="icon-th-large",
							CloseIconHandler=p=>{
								p.Caption="Close icon changed !!!";
								p.CloseIcon.ClassName= "icon-remove-circle";
								p.CloseIconHandler= pn=> {pn.Close(); "Panel was closed".LogError(5000); };
								p.Height="400px";
							}

						});

						panel.Add( new Button(b=>{
							b.Text="click me !";
							b.Style.Width="100%";
							b.Style.Height="100%";
							b.Clicked+= (e) => "button clicked".LogSuccess();

						}));

						panel.Show();


					};
				});


				new Button(d, bt=>{
					bt.Text="Window III";  
					bt.Clicked+=evt=>{  
						var panel= new Panel( new PanelOptions{
							Overlay=true,
							Caption="no closable, no collapsible",
							Left="200px",  
							Top="300px",  
							Width="auto",  
							Collapsible=false,
							Closable=false,
						});

						panel.Add( new Button(b=>{
							b.Text="click me !";
							b.Style.Width="100%";
							b.Style.Height="100%";
							b.Clicked+= (e) => {
								panel.Caption ="now closable and  collapsible";
								panel.Closable=true;
								panel.Collapsible=true;
								b.Disabled=true;
							};

						}));

						panel.Show();


					};
				});


				d.JQuery.Append("C# code".Header(4)).Append(code);
				parent.Append(d);
			});

			  

			var rq =jQuery.GetData<string> ("code/demopanels.html");
			rq.Done (s=> code.InnerHTML=s);

		}

		static Panel CreatePanel(string caption, bool draggable=false, bool closable=false)
		{
			var p = new Panel( new PanelOptions{Draggable=draggable, Closable=closable, Caption=caption});
			p.Body.Style.PaddingTop = "10px";
			p.Body.Style.PaddingBottom = "5px";
			return  p;
		}

		static List<App> GetApps()
		{
			var a = new List<App> (){
				new App{Title="Calculator", Icon="img/calculator.png"},
				new App{Title="Control Panel", Icon="img/control.png"},
				new App{Title="Firewall Settings", Icon="img/firewall.png"},
				new App{Title="Spreadsheet", Icon="img/calc.png"},
				new App{Title="Mail", Icon="img/mail.png"},
				new App{Title="Jack Sparrow Navigator", Icon="img/web.png"},
				new App{Title="MonoDevelop", Icon="img/monodevelop.png"},
				new App{Title="Tomboy", Icon="img/tomboy.png"},
				new App{Title="Skype", Icon="img/skype.png"}
			} ;
			return a;
		}

	}

	public class App:Record
	{
		public App(){}
		public string Title { get; set; }
		public string Icon { get; set; }


	}
	
}

